Src/AutoFixture/RandomNumericSequenceGenerator.cs

```diff
@@ -53,7 +51,17 @@ public RandomNumericSequenceGenerator(params long[] limits)
 
             if (limits.Length < 2)
             {
-                throw new ArgumentException("The limit must be a sequence of two or more integers.");
+                throw new ArgumentException("limits must be two or more Int64.", "limits");
+            }
+
+            //Check that limits are ascending
+            var previous = limits[0];
+            foreach (var item in limits.Skip(1))
+            {
+                if (previous >= item)
+                    throw new ArgumentOutOfRangeException("limits", "limits must be ascending Int64.");
+
+                previous = item;

```

Comment:
Ah, I didn't came across that when I looked for it. Using Zip, that's smart :-)

I'll make the change.